python/pyspark/sql/tests/test_pandas_udf_scalar.py

```diff
@@ -134,6 +134,30 @@ def test_pandas_udf_nested_arrays(self):
         result = df.select(tokenize("vals").alias("hi"))
         self.assertEqual([Row(hi=[["hi", "boo"]]), Row(hi=[["bye", "boo"]])], result.collect())
 
+    def test_pandas_array_struct(self):
+        # SPARK-38098: Support Array of Struct for Pandas UDFs and toPandas
+        import numpy as np
+
+        @pandas_udf("Array<struct<col1:string, col2:long, col3:double>>")
+        def return_cols(cols):
+            self.assertEqual(type(cols), pd.Series)
+            self.assertEqual(type(cols[0]), np.ndarray)
+            self.assertEqual(type(cols[0][0]), dict)
```

Comment:
I wonder if it makes more sense if we can return a NumPy array of DataFrame here.